### PR TITLE
SDESK-5417 add sams assets proxy endpoints

### PR DESCRIPTION
**Pull request by dev-tsoni**

### CODE REVIEW

File: superdesk/sams/assets.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+logger = logging.getLogger(__name__)
+assets_bp = superdesk.Blueprint('sams_assets', __name__)
+
+
+@assets_bp.route('/sams/assets', methods=['GET'])
+def get():
+    """
+    Returns a list of all the registered assets
+    """
+    assets = assets_bp.kwargs['client'].assets.search()
```
Comment (MarkLark86) [inline superdesk/sams/assets.py:40]: We would need to fill in the parameters as passed in from the client (url params)

File: superdesk/sams/assets.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    return response
+
+
+@assets_bp.route('/sams/assets', methods=['POST'])
+def create():
+    """
+    Creates new Asset
+    """
+    files = {'binary': request.files['binary']}
+    docs = request.form.to_dict()
```
Comment (MarkLark86) [inline superdesk/sams/assets.py:79]: Use `request.get_json()` instead

File: superdesk/sams/assets.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    try:
+        etag = request.headers['If-Match']
+    except KeyError:
+        raise SuperdeskApiError.badRequestError(
+            "If-Match field missing in header"
+        )
+
+    files = {'binary': request.files.get('binary')}
+
+    updates = request.form.to_dict()
```
Comment (MarkLark86) [inline superdesk/sams/assets.py:118]: Use `request.get_json()` instead

File: superdesk/sams/assets.py (hunk 1 above)
Comment (dev-tsoni) [reply superdesk/sams/assets.py]: Resolved

File: superdesk/sams/assets.py (hunk 2 above)
Comment (dev-tsoni) [reply superdesk/sams/assets.py:79]: Resolved

File: superdesk/sams/assets.py (hunk 3 above)
Comment (dev-tsoni) [reply superdesk/sams/assets.py:118]: Resolved
request.get_json() returns None for multipart form data.
https://tedboy.github.io/flask/generated/generated/flask.Request.get_json.html.

**Review by MarkLark86 — APPROVED** [review]

**Pull request merged**